### PR TITLE
[WIP] Make disclaimer removal work again

### PR DESCRIPTION
**Pull request by bstoeger**
### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

There was code that removed the old disclaimer from the planner notes. The functionality was unintentionally removed. Reinstate it, but instead of using a global variable with the recently generated disclaimer be smarter and recognize disclaimers with different deco models.
### Changes made:

1) Reinstate removal of dislaimer
2) Remove global variable
### Related issues:

#2264
### Release note:

Will do once finished.
### Documentation change:

No.
### Mentions:

@sfuchs79: Please check this out. It is different from the old code as it only removes the disclaimer, not the whole thing. You know best what to do, so feel free to adapt.

### CODE REVIEW

File: core/planner.h [hunk 1]
```diff
@@ -55,7 +55,6 @@ extern const char *get_planner_disclaimer();
 extern void free_dps(struct diveplan *diveplan);
 extern struct dive *planned_dive;
 extern char *cache_data;
-extern char *disclaimer;
```
Comment (dirkhh) [inline core/planner.h:57]: The build failure on Windows seems to imply that this is still used somewhere in `mainwindow.cpp`

**Comment by bstoeger** [comment]
Yes, for printing. Obviously this was dysfunctional, as the global variable wasn't set. I will have to look into this when I have a clear mind.

**Comment by dirkhh** [comment]
Ah, and this one needs some attention as well :-)

**Review by atdotde — CHANGES_REQUESTED** [review]
The point of the old code was (and at some point that worked) not so much to remove the disclaimer but actually the old dive plan but not additional stuff that had been manually entered above the plan. The disclaimer was only used as the anchor to find the start of the dive plan (before the disclaimer text got the %s). So I would chop off everything from disclaimerPositionStart.

**Comment by bstoeger** [comment]
OK - changed it in that way. Completely untested, please check it out.

Also removed the global disclaimer variable from the plan-printing routine. Is that still needed?

**Comment by atdotde** [comment]
Hmm, I liked the printed plan having a disclaimer (even if it's just snake oil). The complication with the mentioning of the deco algorithm together with our mixture of C/Qt with %s vs. %1 for .arg() are unfortunately preventing an easy solution. What do you think?

**Comment by bstoeger** [comment]
Ok - added a function that returns disclaimer with current deco-mode.

Please test.

File: core/planner.h (hunk 1 above)
Comment (bstoeger) [reply core/planner.h:57]: Should be fixed.

**Comment by atdotde** [comment]
Looks, good. I restarted the Mac build. One could now worry that the plan is not removed when upon replanning the model is switched between VPM-B an Bühlmann (to prevent that only match the first bit of the disclaimer), but I am happy with the current state.

**Review by atdotde — APPROVED** [review]

**Comment by bstoeger** [comment]
> One could now worry that the plan is not removed when upon replanning the model is switched between VPM-B an Bühlmann (to prevent that only match the first bit of the disclaimer)

I don't understand: In contrast to the old code, the new code matches beginning and end of the disclaimer, but ignores the deco-type. Now it should work when changing the model as opposed to the old code, no?

**Comment by atdotde** [comment]
You are absolutely right. I only read a singe commit and did not realise you now have two functions, one with %s and one with the actual deco model string.

**Pull request merged**